app/src/main/java/org/simple/clinic/remoteconfig/firebase/FirebaseRemoteConfigService.kt

```diff
@@ -2,34 +2,28 @@ package org.simple.clinic.remoteconfig.firebase
 
 import com.google.firebase.remoteconfig.FirebaseRemoteConfig
 import io.reactivex.Completable
+import org.simple.clinic.platform.crash.CrashReporter
 import org.simple.clinic.remoteconfig.ConfigReader
 import org.simple.clinic.remoteconfig.RemoteConfigService
 import org.simple.clinic.util.toCompletable
-import org.threeten.bp.Duration
 import timber.log.Timber
 import javax.inject.Inject
 
 class FirebaseRemoteConfigService @Inject constructor(
     private val firebaseRemoteConfig: FirebaseRemoteConfig,
-    /**
-     * Duration to cache local values before they're synced with Firebase again.
-     * This is required because Firebase throttles network calls by returning cached
-     * values once the limit is reached. More information here:
-     *
-     * https://firebase.google.com/docs/remote-config/android
-     */
-    private val cacheExpirationDuration: Duration
+    // This is marked as Lazy to avoid a cyclic dependency between FirebaseRemoteConfigService and
+    // SentryCrashReporter (which happens only in release builds).
+    // TODO(vs): 2020-01-17 Change CrashReporter to be something like Timber, which can live outside the DI graph
+    private val crashReporter: dagger.Lazy<CrashReporter>
 ) : RemoteConfigService {
 
   override fun reader(): ConfigReader = FirebaseConfigReader(firebaseRemoteConfig)
 
   override fun update(): Completable {
     return firebaseRemoteConfig
-        .fetch(cacheExpirationDuration.seconds)
+        .fetchAndActivate()
         .toCompletable { Timber.w("Failed to update Firebase remote config") }
```

Comment:
Let's change it to this? Because right now, it looks like, when task completes it means that the firebase remot econfig failed to update, which is not the case.
`.toCompletable(onTaskUnsuccessful = { Timber.w("Failed to update Firebase remote config") })`

Reply:
Good point, I'll make this change.